components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/association/federation/FederatedAssociationManagerImpl.java

```diff
@@ -467,4 +478,31 @@ private String getIdentityProviderName(String tenantDomain, String idpId)
                     null, true);
         }
     }
+
+    private String getIdentityProviderNameByResourceId(String tenantDomain, String idpResourceId)
+            throws FederatedAssociationManagerException {
+
+        try {
+            IdpManager idpManager = IdentityUserProfileServiceDataHolder.getInstance().getIdpManager();
+            if (idpManager == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("The IdpManager service is not available in the runtime");
+                }
+                String msg = "Error while retrieving identity provider for the federated association";
+                throw new FederatedAssociationManagerException(msg);
+            }
+            IdentityProvider identityProvider = idpManager.getIdPByResourceId(idpResourceId, tenantDomain, false);
```

Comment:
Rather than fetching the whole IDP object just to get the name from UUID, wouldn't it be better to use (if not introduce) a method that simply returns the name of the IDP when resourceID is provided.

Also resourceId is supposed to be unique across tenants so we should be simply be able to use the UUID to get the IDP name

Reply:
Fixed in 9fee8a6